core/workspace_svg.js

```diff
@@ -993,6 +1024,11 @@ Blockly.WorkspaceSvg.prototype.resize = function() {
   if (this.scrollbar) {
     this.scrollbar.resize();
   }
+  var uiPlugins = this.plugins['uiplugin'];
+  for (var i = 0; uiPlugins && i < uiPlugins.length; i++) {
+    var plugin = uiPlugins[i];
+    plugin.position();
+  }
```

Comment:
This is interesting, because only plugins that are registered as type `uiplugin` will be positioned here. In the case of the Toolbox, it's of type 'toolbox' and 'uiplugin' (given both the inheritance tree in the interface, and the fact that it implements `position()`), so perhaps what you're really after here, is get me all registered components that implement position?
If so, we should think about how the registry can solve for that.

Reply:
I am not convinced I would want to get all registered components that implement position. It seems like that can get messy if people have methods on their classes that they don't realize are part of a different interfaces.

Reply:
Well there are other ways to tell if a class implements a specific interface. Unfortunately JS doesnt have that built in, but perhaps that interface can require a property such as:
`Blockly.UIPlugin.prototype.IS_UI_PLUGIN = true` and all plugins that have that bit set to true will also have position implemented (as we know they implement the UIPlugin interface and hence can safely call position on them)

Reply:
Ah, yeah that makes sense. We can potentially use that information for figuring out what lists to add it to on the workspace?

Reply:
I was thinking more along the lines that all plugins are in one list. You can use a property like that to filter for plugins that implement a specific interface.